types/timestamp.go

```diff
@@ -31,3 +31,17 @@ func (ts TimestampSlice) Less(i, j int) bool {
 func (ts TimestampSlice) Swap(i, j int) {
 	ts[i], ts[j] = ts[j], ts[i]
 }
+
+// Clock allows clients to retrieve the current time.
+type Clock interface {
```

Comment:
as long as it's in the same package, you don't need it to be exported. Exporting is for leaving the package, not leaving the file.